### PR TITLE
Replacing `cached_path` by `ureq`

### PR DESCRIPTION
**Pull request by robertbastian**
`cached_path` pulls in `reqwest`, which is dependency-heavy. `ureq` is a minimal HTTP library that removes 40 crates from our `Cargo.lock`.

### CODE REVIEW

File: tools/testdata-scripts/src/bin/download-repo-sources.rs [hunk 1]
```diff
@@ -57,10 +58,26 @@ fn main() -> eyre::Result<()> {
...
     let output_path = &args.out;
 
-    let cached = cached_path::CacheBuilder::new()
-        .freshness_lifetime(u64::MAX)
-        .build()
-        .unwrap();
+    fn cached(resource: &str) -> Result<PathBuf, DataError> {
+        let root = std::env::var_os("ICU4X_SOURCE_CACHE")
+            .map(PathBuf::from)
+            .unwrap_or_else(|| std::env::temp_dir().join("icu4x-source-cache/"))
```
Comment (sffc) [inline tools/testdata-scripts/src/bin/download-repo-sources.rs:64]: Question: should we go back to using `dirs::cache_dir()`?

**Review by sffc — COMMENTED** [review]
Seems good; leaving it up to Manish to sign off on the deps

**Review by Manishearth — APPROVED** [review]
It's by algesten and jsha, I trust it, and love the reduction in deps.

File: tools/testdata-scripts/src/bin/download-repo-sources.rs (hunk 1 above)
Comment (robertbastian) [reply tools/testdata-scripts/src/bin/download-repo-sources.rs:64]: What's the advantage? It's an extra dependency.

Comment (sffc) [reply tools/testdata-scripts/src/bin/download-repo-sources.rs:64]: I assume `cache_dir` is more likely to be persisted, so better for a "cache", whereas `/tmp` can be deleted at any time

Comment (robertbastian) [reply tools/testdata-scripts/src/bin/download-repo-sources.rs:64]: I think it's good if it gets cleared occasionally, with `cache_dir` we might get complaints that we're not cleaning up after ourselves :)

**Pull request merged**